src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -21,11 +21,23 @@ namespace ILCompiler
 {
     public class CompilationOptions
     {
-        public IReadOnlyDictionary<string, string> InputFilePaths;
-        public IReadOnlyDictionary<string, string> ReferenceFilePaths;
+        /// <summary>
+        /// Associate for each simple name, the list of assembly full path names matching the simple name.
+        /// Useful when including multiple assemblies with same name from different locations.
```

Comment:
Agree, this can be error or warning - to detect issues early. 

I believe that we depend on the current behavior (pick the first hit) in several places today. If we make this hard error, there is cleanup to be done.